### DISCUSSION TITLE
Keeping track of my GPG keys?

### DISCUSSION THREAD

**Discussion by applejag** [community/community#22352] [New to GitHub]
<p>Hi community!</p>
<p><strong>Short:</strong> Tips on knowing which GPG key is which on my Github settings page?</p>
<p><strong>Long:</strong> I’m reguarly switching computers. I have always two or more in the spin at a single time. (Home desktop, work desktop, work laptop…), and I develop on all of these.</p>
<p>I’m in the habit of signing my work via GPG keys. Different keys for different computers (and yes different passphrases for each key, calm down).</p>
<p>But my GPG settings page (<a href="https://github.com/settings/keys" rel="nofollow noopener">https://github.com/settings/keys</a>) gets very cluttered with my keys. And the list only specifies the Key ID and email address assigned.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/f90a9cc72ec02cdc08d6221c8124de7954eb3a36.png" data-download-href="/uploads/short-url/zx7CSHIXHJUoqsUb5wKwYIrUZbE.png?dl=1" title="gpg which key.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/2477952/181070296-d82cc702-03cd-47be-8059-bbd6ffce58c3.png" alt="gpg which key.png" data-base62-sha1="zx7CSHIXHJUoqsUb5wKwYIrUZbE" width="690" height="232" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/f90a9cc72ec02cdc08d6221c8124de7954eb3a36_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">gpg which key.png</span><span class="informations">745×251 18.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Problem arises when I want to remove an old key, from a computer I’ve already cleansed. “<em>Which one of my keys was on that computer now again?</em>” -not too regular of an occurence, but a huge annoyance when it does happen, for I have to use the exclusion method on all my keys on all my computers to determine which key it was, before I can finally remove it from my Github settings.</p>
<p><strong>My question is:</strong> Any goodie tips for avoiding this?</p>
<p>Is it good/bad practice to move around the same key everywhere and not use different keys? (like via USB? is that possible?) Feels safer to use distinct keys and discarding them when finished, but Github doesn’t show the key comment in the settings page. :frowning:</p>
<p>Thanks in advance!</p>

**Comment by mpboom** [comment, marked as answer]
<p>First of all, I am wondering why you are deleting your keys instead of securely destroying them? I always believed that the commits that are signed with a key that is not added to your account will show up as “unverified”.</p>
<p>Personally, I think it’s a good idea to just use one key and move it around on some secure device like a <a href="https://www.yubico.com/" rel="nofollow noopener">YubiKey</a> for example.</p>
<p>But there are a lot of reasons why that wouldn’t be possible. Keeping a little spreadsheet/document with the key ID mapped to the description might be the best solution then.</p>

**Comment by applejag** [comment]
<p>Oh my! You’re totally right!</p>
<p>I thought once verified commit, always verified commit.</p>
<p>Now I have lots of repos with good size quantities of “Unverified” commits. Good job me.</p>
<p>Yea I’m just gonna keep them instead of deleting them then, and spreadsheet the keys just in case I need to know which one’s which someday.</p>
